### ISSUE TITLE
App will not automatically update, errors out.

### ISSUE THREAD

**Issue by TobiWtf** [electron-userland/electron-builder#5827]
* **Version 22.10.5**: 

* **Electron Version 12.0.5**:
* **Electron Type (current)**:

* **Electron builder 4.3.8**

* **Target windows**: 

So, the app starts the update, but it never reaches the event "update-downloaded", it never gets there. I am not sure why, but, i am able to view the download progress, it finishes, but never emits the event, or actually downloads the update.

Here is my code for my main process

```js
const { app, BrowserWindow, ipcMain, Notification } = require("electron")
const { autoUpdater } = require('electron-updater')

const windows = {}

app.on("ready", () => {

    windows.updater = new BrowserWindow({
        width: 310,
        height: 390,
        fullscreenable: false,
        frame: false,
        webPreferences: {
            nodeIntegration: true,
            contextIsolation: false,
        }
    })

    windows.updater.loadFile("./src/screens/updater/index.html")

    autoUpdater.checkForUpdatesAndNotify()

    new Notification({
        body: "Test notification"
    }).show()
})

app.on("window-all-closed", () => {
    if (process.platform !== "darwin") {
        app.quit()
    }
})

ipcMain.on("get-app-version", event => {
    event.sender.send("app-version", { version: app.getVersion() })
})

autoUpdater.on("update-available", () => {
    new Notification({
        body: "an update is available to your app, it will start downloading"
    }).show()
})



autoUpdater.on("update-downloaded", () => {
    new Notification({
        body: "an update has been downloaded to your app, it will now close and install."
    }).show()

    autoUpdater.quitAndInstall()
})

autoUpdater.on("update-not-available", () => {
    new Notification({
        body: "App is up to date!!"
    }).show()
})

autoUpdater.on("error", err => {
    new Notification({
        body: err.message
    }).show()
})

autoUpdater.on('download-progress', (progressObj) => {
    let log_message = "Download speed: " + progressObj.bytesPerSecond;
    log_message = log_message + ' - Downloaded ' + progressObj.percent + '%';
    log_message = log_message + ' (' + progressObj.transferred + "/" + progressObj.total + ')';
    new Notification({
        body: log_message
    }).show()
})
```

Any help is appreciated, i have been raging for the last four hours because for every fix i test on my own, it requires me to publish the fix, then publish the same fix but with a new version number, and then open the fix file, and it has become very tedious and irritating and i need some fresh sets of eyes to check it out. I can link to the source codes repo if anyone needs.

**Comment by TobiWtf** [comment]
Update: after ten minutes, it said `net::ERR_CONNECTION_CLOSED` so it never even makes it past the download request to github

**Comment by connectian1** [comment]
Did you find any solution? I have the same problem. It gives me this error:
"Cannot download differentially, fallback to full download: Error: Maximum allowed size is 50 MB". 
It is getting frustrating

**Comment by theheX789** [comment]
i have same issue after update electron to v.12

**Comment by TobiWtf** [comment]
> Did you find any solution? I have the same problem. It gives me this error:
> "Cannot download differentially, fallback to full download: Error: Maximum allowed size is 50 MB".
> It is getting frustrating

So that day after a few hours I installed `electron-log` and got the same error, eventually I downloaded an older version of multiple electron tools and I (think?) Electron, [here](https://github.com/ifunny-pc/ifchat) is the repo I was working with, you can download the versions of stuff from that package.json, hopefully that works. I really wish electron builder and all of these tools would actually test their software with newer versions of stuff before releasing to production.

There is no fix to this it appears. I can do a check later to make sure, but I don't think there is a fix. Thankfully, I believe I was still able to use a version of v.12, hopefully this helps.

**Comment by TobiWtf** [comment]
> i have same issue after update electron to v.12

https://github.com/electron-userland/electron-builder/issues/5827#issuecomment-828993326

**Comment by TobiWtf** [comment]
> fallback

So, i took a look into electron builder, this is the statement that is causing issues in electron builder

`node_modules\builder-util-runtime\out\httpExecutor.js` At line 232 has this function

```js
        responseHandler: (response, callback) => {
          const contentLength = safeGetHeader(response, "content-length");
          let position = -1;

          if (contentLength != null) {
            const size = parseInt(contentLength, 10);

            if (size > 0) {
              if (size > 52428800) {
                callback(new Error("Maximum allowed size is 50 MB"));
                return;
              }

              result = Buffer.alloc(size);
              position = 0;
            }
          }

          response.on("data", chunk => {
            if (position !== -1) {
              chunk.copy(result, position);
              position += chunk.length;
            } else if (result == null) {
              result = chunk;
            } else {
              if (result.length > 52428800) {
                callback(new Error("Maximum allowed size is 50 MB"));
                return;
              }

              result = Buffer.concat([result, chunk]);
            }
          });
          response.on("end", () => {
            if (result != null && position !== -1 && position !== result.length) {
              callback(new Error(`Received data length ${position} is not equal to expected ${result.length}`));
            } else {
              callback(null);
            }
          });
        }
```
        
Specifically, this here is causing issues

```js
              if (result.length > 52428800) {
                callback(new Error("Maximum allowed size is 50 MB"));
                return;
              }
```
And this statement here
```js
            if (size > 0) {
              if (size > 52428800) {
                callback(new Error("Maximum allowed size is 50 MB"));
                return;
              }

              result = Buffer.alloc(size);
              position = 0;
            }
```

I personally done have the time to test it right now, but if anyone wants to try removing that statement, be my guest, make a fork haha.

Who in there right mind thought giving electron apps a limit of 50mb was a good idea? Who knows, it should really have a limit of like, 150-300 mb. Lmk if you guys decide to test that.

**Comment by TobiWtf** [comment]
> i have same issue after update electron to v.12

https://github.com/electron-userland/electron-builder/issues/5827#issuecomment-829000152

**Comment by TobiWtf** [comment]
> Did you find any solution? I have the same problem. It gives me this error:
> "Cannot download differentially, fallback to full download: Error: Maximum allowed size is 50 MB".
> It is getting frustrating

https://github.com/electron-userland/electron-builder/issues/5827#issuecomment-829000152

**Comment by TobiWtf** [comment]
Going to leave one last note, having a 50mb limit is dumb because 3/5th's of that is used by electron alone, electron is like 33mb, add in any other packages, a few images, anything, and you'll be over that limit very quickly. Honestly, the limit should not exist, theres no reason to have a limit to how big updates should be.

**Comment by connectian1** [comment]
I tried with the package version you are using, download starts but does not end.
Yeah, I had tried commenting the statement causing the error, it works without it because it downloads the exe. 
But still it doesn't sit well with me because it should download .blockmap and not .exe. So the problem is deeper here.
I hope they are going to find a solution. I wonder how the others are doing, since it doesn't work
